news-explorer-frontend/src/components/App/App.jsx

```diff
@@ -0,0 +1,342 @@
+            setSavedArticle(savedArticlesFiltered)
+          })
+      }
+    }, [currentUser])
+
+    //MainApi getUserInfo getArticles
+    useEffect(() => {
+      const token = localStorage.getItem('jwt')
+      if(token) {
+        api.getUserInfo(token)
+          .then( res => {
+            setCurrentUser(res.data);
+          })
+          .catch((err) => console.log("getUserInfo=>", err));
+        api.getArticles(token)
+          .then(res => {
+            setCards(res.data);
+          })
+          .catch((err) => console.log("getArticles=>" , err));
+      } 
+    }, [isLoggedIn]) //dependency loggedin eklendi
+
+    
+    function bookmarkCard({ _id , keyword, title, text, source, date, publishedAt, link, url, description, image, urlToImage, }) {
+      //  const card = {keyword: source.name , title, text: description , source: source.name, date: publishedAt, link: url , image: urlToImage, _id };
+      //  const currentCard = card;
+      console.log("currentUser._id", currentUser._id);
+      api
+        .saveArticle({ _id, keyword: source.name , source: source.name, title, text: description, date: publishedAt, link: url, image: urlToImage,  })
+          .then((card) => {  
+            const { data } = card;
+            setSavedArticle([...savedArticle, data]); //setSavedArticle([...savedArticle, card]);
+            console.log("card-id", data._id);
+          })
+          //.then(console.log("savedArticle =>" , savedArticle))
+          .catch((err) => console.log("bookmark Error =>", err));
+    }
+
+    // useEffect(() =>  {
+    //   console.log("savedArticle =>" , savedArticle)
+    // }, [savedArticle])
+
+     //remove bookmark
+    function deleteCardFromSaved(card) {
+      api
+        .deleteArticle(card.id)
+          .then((res) => {
+            console.log("deleted card props.id ...>", res._id )
+            const newSavedArticles = savedArticle.filter((card) => card._id !== res._id);
+          setSavedArticle(newSavedArticles);
+          console.log("savedArticles==>", savedArticle);
+
+        })
+        .catch((err) => console.log("delete function error =>", err))
+    }
+
+    function searchItems(searchValue) {
+      if(searchValue !== '') {  // console.log(searchValue);
+        setIsLoading(true);
+    
+          newsApi.getNews(searchValue)
+            .then( (res) => {   
+              console.log("res--->", res)
+              let searchResult = res.articles.filter((item) => {
+                return Object.values(item)
+                  .join('')
+                  .toLowerCase()
+                  .includes(searchValue.toLowerCase());
+              })
+              setFilteredResults(searchResult)
+            })
+            .finally(() => {
+              setIsLoading(false);
+            })          
+      } else { 
+        console.log("nothing found....");
+        setFilteredResults('')
+      }
+    }
+
+    useEffect(() => {
+      searchItems(keyword);
+    }, [keyword]);

```

Comment:
please, delete this effect because it will be launched every time the `keyword` is changed. It's not needed. 

Only when the search form is submitted it should be done

Reply:
fixed, called the searchItems functions within the SearchForm component under the function handleSearchSubmit - then changed searchresults - ln43 with this - so all is fine after a search done it either shows result or NotFound component
 (            keyword.length > 0 &&  filteredResults &&  <NotFound/>      )